### ISSUE TITLE
Migrate to Clerk Core 2 and add new styled components

### ISSUE THREAD

**Issue by thatbeautifuldream** [thatbeautifuldream/ai-roadmap-generator#40]
Clerk recently released its version update with multiple improvements specifically middleware logic.

Now by default all routes are public and can be protected manually, which is a better approach as our OG images on share roadmap route were failing to generate because of the exact issue as the middleware was getting triggered and not letting the request pass through.

**Comment by temideewan** [comment]
What routes should now be protected? Looks like it should be an inverse of the current array of publicly available routes yeah?

looking to use the new [clerkMiddleware function](https://clerk.com/docs/references/nextjs/clerk-middleware)

**Comment by thatbeautifuldream** [comment]
Looking to protect /dashboard and /roadmap

**Comment by temideewan** [comment]
Hi @thatbeautifuldream can I take this on?

**Comment by thatbeautifuldream** [comment]
Yes surely.

**Comment by temideewan** [comment]
Hi @thatbeautifuldream I think I'd have to update the clerk dependency to get access to the helper functions specified in the documentation

**Comment by thatbeautifuldream** [comment]
> Hi @thatbeautifuldream I think I'd have to update the clerk dependency to get access to the helper functions specified in the documentation

yes feel free to try it out locally first and the migration tool as well.

**Comment by temideewan** [comment]
Ohhh alright then
